src/components/app.tsx

```diff
@@ -38,54 +40,62 @@ export const App: React.FC<AppProps> = ({ className, model }) => {
     });
 
     return (
-        <main className={st(classes.root, className)}>
+        <main className={st(classes.root, { hasError: Boolean(error) }, className)}>
             <Header className={classes.header} />
-            <FileExplorer
-                filePaths={Object.keys(files)}
-                onSelect={setSelected}
-                onRemove={removeFile}
-                onAddFile={addFile}
-                selected={selected}
-                className={classes.fileExplorer}
-            />
-            <section className={classes.editors}>
-                <Pane title="Source" className={classes.source}>
-                    <Editor
-                        value={files[selected]}
-                        onChange={(value) => updateSelected(value)}
-                        language={selected.endsWith('.st.css') ? 'css' : 'javascript'}
-                        options={{ minimap: { enabled: false }, scrollBeyondLastLine: false }}
-                        onMount={resizeEditor}
-                    />
-                </Pane>
-                <Pane title="Target" className={classes.target}>
-                    <Editor
-                        value={formattedOutput}
-                        language="css"
-                        options={{
-                            readOnly: true,
-                            minimap: { enabled: false },
-                            scrollBeyondLastLine: false,
-                        }}
-                        onMount={resizeEditor}
-                    />
-                </Pane>
-                <Pane title="Meta" className={classes.ast}>
-                    <Editor
-                        value={printedMeta}
-                        language="json"
-                        options={{
-                            readOnly: true,
-                            minimap: { enabled: false },
-                            scrollBeyondLastLine: false,
-                        }}
-                        onMount={resizeEditor}
-                    />
-                </Pane>
-                <Pane title="Diagnostics" className={classes.diagnostics}>
-                    <DiagnosticsList diagnostics={diagnostics} meta={meta} />
-                </Pane>
-            </section>
+            {
+                !error ? (
```

Comment:
we can do better....when there is an error, only small part of the app is affected so it will be better to show the error somewhere in the UI (with the ability to collapse) and recover from the error at the throw point.

Reply:
Also if you warp the entire render with if it makes it less readable there are better solutions for that....
1. if component
2. branch also the outer jsx
3. **different error solution**